### ISSUE TITLE
When parsing JSONata, where can regular expressions appear?

### ISSUE THREAD

**Issue by cristiangraz** [jsonata-js/jsonata#264]
I am trying to determine how to differentiate a division operator from the start of a regex in a JSONata lexer/parser.

The test suite shows them only used as function parameters. I know you can also use them with the function chaining operator `~>` because the regex returns a function. Are those the only two valid use cases? Any tips/suggestions

**Comment by andrew-coleman** [comment]
As far as the parser is concerned, `/` is interpreted as a division operator if it is parsed as a binary operator, and treated as the start of a regex if parsed as a unary operator.

There is a discussion on regex handling in [this issue](https://github.com/jsonata-js/jsonata/issues/177).  Hopefully that will help with the second part of your question.

**Comment by cristiangraz** [comment]
Thanks @andrew-coleman. The binary = division, unary = regex makes sense. I saw the JS implementation has a `prefix` (boolean?) passed to the `next()` function that directs it to either a regex (if `prefix !== true`) or division operator.

I'm trying to determine the correct logic in a JSONata lexer/parser where the parser can know if it's a prefix or a binary expression before the lexer grabs the next token. I tried digging into the source but couldn't quite make out the exact logic that you are currently using to make that determination.
